DependencyInjection/CompilerPass/RemoveTaggedServiceCollectorCompilerPass.php

```diff
@@ -0,0 +1,58 @@
+<?php
+ 
+namespace Sulu\Bundle\FormBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class RemoveTaggedServiceCollectorCompilerPass implements CompilerPassInterface
```

Comment:
as this is a compilerpass inside the form bundle an can't be used as generic sulu/sulu compilerpass I would avoid any constructor parameters and for transparency name what it is build for:

```php
class DynamicDisabledTypesCompilerPass  implements CompilerPassInterface
{
    public function process(ContainerBuilder $container)
    {
        $disabledSerivcesAliases = $container->getParameter('sulu_form.dynamic_disabled_types');
        $taggedServices = $container->findTaggedServiceIds('sulu_form.dynamic.typ');

        foreach ($taggedServices as $id => $attributes) {
            if(isset($attributes[0]['alias']) && in_array($attributes[0]['alias'],$disabledSerivcesAliases)) {
                if ($container->hasDefinition($id)) {
                    $container->getDefinition($id)->setTags([]); // remove this if its not longer needed
                    $container->removeDefinition($id);
                }                
            }
        }
    }
}
```

Reply:
@rflorent ignore this comment as we could maybe reuse it internal in the form bundle for the other tag service!